for-research/phoenix/get_json_phoenix.py

```diff
@@ -32,159 +32,243 @@
 # Opening the ntuple and reading ALL of the branches will take unnecessary time and memory
 # Let's define a schema to only use the branches we care about
 
+# Helper function to create a collection from branch forms
+def create_collection(branch_forms, possible_branches, collection_name, vector_type=None, additional_branches=None):
+    # Initialize a dictionary to store the branches we will collect
+    branch_dict = {}
+
+    # Loop over possible branches and check if they exist in the input branch forms
+    # If they do, add them to the branch_dict
+    for key, branch_name in possible_branches.items():
+        if branch_name in branch_forms:
+            branch_dict[key] = branch_forms[branch_name]
+
+    # If additional branches are provided, check if they exist and add them as well
+    if additional_branches:
+        for key, branch_name in additional_branches.items():
+            if branch_name in branch_forms:
+                branch_dict[key] = branch_forms[branch_name]
+
+    # If we found any valid branches, return a zipped form (possibly with a vector type)
+    if branch_dict:
+        if vector_type:
+            return zip_forms(branch_dict, collection_name, vector_type)
+        else:
+            return zip_forms(branch_dict, collection_name)
+    else:
+        # If no branches were found for the collection, print a message and return None
+        print(f"No branches found for {collection_name}, skipping {collection_name}.")
+        return None
+
 class PHYSLITE_NtupleSchema(BaseSchema):
     def __init__(self, base_form):
         super().__init__(base_form)
         self._form['contents'] = self._build_collections(self._form['contents'])
-
+    
```

Comment:
This looks like an editor setting? You might see if you can turn it off to avoid injecting white space where it isn't needed...